components/support/utils/src/test/java/mozilla/components/support/utils/SafeIntentTest.kt

```diff
@@ -4,23 +4,284 @@
+
+        category.add("NOT" + Intent.CATEGORY_LAUNCHER)
+
+        doReturn(category)
+                .`when`(intent).categories
+
+        doReturn("NOT" + Intent.ACTION_MAIN)
+                .`when`(intent).action
+
+        assertFalse(SafeIntent(intent).isLauncherIntent)
+
+        // action is not valid
+        category.clear()
+
+        category.add(Intent.CATEGORY_LAUNCHER)
+
+        doReturn(category)
+                .`when`(intent).categories
+
+        doReturn("NOT" + Intent.ACTION_MAIN)
+                .`when`(intent).action
+
+        assertFalse(SafeIntent(intent).isLauncherIntent)
+
+
+        // category is not valid
+        category.clear()
+
+        category.add("NOT" + Intent.CATEGORY_LAUNCHER)
+
+        doReturn(category)
+                .`when`(intent).categories
+
+        doReturn(Intent.ACTION_MAIN)
+                .`when`(intent).action
+
+        assertFalse(SafeIntent(intent).isLauncherIntent)
+
+
+        // both are valid
+        category.clear()
+
+        category.add(Intent.CATEGORY_LAUNCHER)
+
+        doReturn(category)
+                .`when`(intent).categories
+
+        doReturn(Intent.ACTION_MAIN)
+                .`when`(intent).action
+
+        assertTrue(SafeIntent(intent).isLauncherIntent)
+    }
+
+
+    @Test
+    fun `isLauncherIntent returns true if intent is Launcher Intent`() {
+
+        //both category and action are not valid
+        val category = HashSet<String>()
+        category.add(Intent.CATEGORY_LAUNCHER)
+
+        doReturn(category)
+                .`when`(intent).categories
+
+        doReturn(Intent.ACTION_MAIN)
+                .`when`(intent).action
+
+        assertTrue(SafeIntent(intent).isLauncherIntent)
+    }
+
+
+    @Test
+    fun `getDataString returns null if intent throws OutOfMemoryError`() {
+
+        doThrow(OutOfMemoryError::class.java)
+                .`when`(intent).dataString
+
+        assertNull(SafeIntent(intent).dataString)
+    }
+
+    @Test
+    fun `getData returns null if intent throws OutOfMemoryError`() {
+
+        doThrow(OutOfMemoryError::class.java)
+                .`when`(intent).data
+
+        assertNull(SafeIntent(intent).data)
+    }
+
+    @Test
+    fun `getCategories returns null if intent throws OutOfMemoryError`() {
+
+        doThrow(OutOfMemoryError::class.java)
+                .`when`(intent).categories
+
+        assertNull(SafeIntent(intent).categories)
+    }
+
+    @Test
+    fun `hasExtra returns false if intent throws OutOfMemoryError`() {
+
+        doThrow(OutOfMemoryError::class.java)
+                .`when`(intent).hasExtra(anyString())
+
+        assertFalse(SafeIntent(intent).hasExtra(""))
+    }
+
+    @Test
+    fun `getBooleanExtra returns false if intent throws OutOfMemoryError`() {
+
+        doThrow(OutOfMemoryError::class.java)
+                .`when`(intent).getBooleanExtra(anyString(), anyBoolean())
+
+        assertFalse(SafeIntent(intent).getBooleanExtra("", false))
+    }
+
+
```

Comment:
\[ktlint\] WARNING: \[no\-consecutive\-blank\-lines\] Needless blank line\(s\) in column 1